### PR TITLE
Issue #745 localized trainer names trainer classes and titles

### LINKED ISSUE #745: Adding localization to trainer names and titles
I noticed that currently all of the names and dialogue is english only.

I want to add a way to localize them and also provide german localization (the only language i speak of the supported ones besides english).

I will work on this issue. It is there so other can give feedback/help etc

### PR DESCRIPTION
**Pull request by CodeTappert**
Fixes #745 

Issue #745 - Added the option to localize titles, trainer names (for important trainers like elite 4, champs etc) and trainer classes.

- Also i already did the whole localization for german (sorry thats the only language is speak other then english...)
- Also renamed the trainer Type "student" to school_kid (because there apparently really is a trainer class called student since the gen 9 dlc)
- And i changed it so it makes sure that i18 only gets initalized once (it may be needed to initalize it before the loading phase so the elite 4 titles etc can be localized)

### CODE REVIEW

**Comment by Adri1** [comment]
About traner class divded by gender, are those you doing or it was already like this?
Because if it was your dosing, adding a female version for "Rival" class and "Champion" class would be perfect for languages like French

**Comment by Adri1** [comment]
French translation submitted to your fork

**Comment by CodeTappert** [comment]
> About traner class divded by gender, are those your doing or it was already like this? Because if it was your doing, adding a female version for "rival", "champion", "professor" and "frontier_brain" classes would be perfect for languages like French

Well in regards to the "titles" i used what was already there in pokerogue. Only for gym leaders i decided to split it (but it isnt used).

For everything else (trainer classes) i went to https://bulbapedia.bulbagarden.net/ to see what is written there for trainer classes.

So the split would need to implemented by ourselfs

**Comment by CodeTappert** [comment]
ALso i merged your pull request so the french translation is added

**Comment by Adri1** [comment]
Also do not forget to check the reviews I made!

**Comment by CodeTappert** [comment]
> Also do not forget to check the reviews I made!

Sadly all of them are just changes in localization strings of languages i dont speak... So i can not really help!

**Comment by Adri1** [comment]
Even the typo for "hex_maniac" class written "hex_manic"?

**Comment by CodeTappert** [comment]
> Even the typo for "hex_maniac" class written "hex_manic"?

Could you maybe link to what you mean? I am not a 100% sure what you are talking about

**Comment by Adri1** [comment]
> > Even the typo for "hex_maniac" class written "hex_manic"?
> 
> Could you maybe link to what you mean? I am not a 100% sure what you are talking about

Line 56 in trainers.ts
image
While in the rest of the code, it's properly written "hex_maniac"

**Comment by CodeTappert** [comment]
OKay i added the fix.

**Comment by CodeTappert** [comment]
> > > Even the typo for "hex_maniac" class written "hex_manic"?
> > 
> > 
> > Could you maybe link to what you mean? I am not a 100% sure what you are talking about
> 
> Line 56 in trainers.ts files of all languages image While in the rest of the code, it's properly written "hex_maniac" (here in trainer-names.ts) image

But i still not see any reivew comments you made here on github in this pull request

**Comment by CodeTappert** [comment]
Would still love to see a review/merge on this.

**Review by Adri1 — APPROVED** [review]

File: src/locales/en/trainers.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    "parasol_lady": "Parasol Lady",
+    "pilot": "Pilot",
+    "pokefan": "Pokefan",
+    "preschooler": "Preschooler",
+    "preschooler_female": "Preschooler",
+    "psychic": "Psychic",
+    "psychic_female": "Psychic",
+    "ranger": "Ranger",
+    "rich": "Gentleman", // Gentleman is the english name but the trainerType is rich
+    "rich_kid": "Rich Kid",
```
[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/en/trainers.ts:69]: It is literally defined as "Gentleman" in the code. (it links to the gentleman trainerclass in the code)

[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/en/trainers.ts:69]: See trainer-names.ts line 57

`  [TrainerType.RICH]: new TrainerNameConfig(TrainerType.RICH, 'Gentleman').hasGenderVariant('Madame'),
`

[the remark this answers is not in the document]
Comment (Adri1) [reply src/locales/en/trainers.ts:69]: I'm talking about "rich_kid" on line 69, not the Gentleman

[the remark this answers is not in the document]
Comment (CodeTappert) [reply src/locales/en/trainers.ts:69]: Oh well then you are correct indeed.

**Comment by CodeTappert** [comment]
> Would still love to see a review/merge on this.

.

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**